packages/core-backend/src/AccountActivityService.test.ts

```diff
@@ -0,0 +1,2936 @@
+
+      // AccountActivityService catches errors and forces reconnection instead of throwing
+      await accountActivityService.subscribeAccounts(mockSubscription);
+
+      // Should have attempted to force reconnection
+      expect(messengerMocks.disconnect).toHaveBeenCalled();
+      expect(messengerMocks.connect).toHaveBeenCalled();
+    });
+
+    it('should handle account activity messages', async () => {
+      let capturedCallback: (notification: ServerNotificationMessage) => void =
+        jest.fn();
+
+      // Mock the subscribe call to capture the callback
+      messengerMocks.connect.mockResolvedValue(undefined);
+      messengerMocks.disconnect.mockResolvedValue(undefined);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      messengerMocks.subscribe.mockImplementation((options: any) => {
+        // Capture the callback from the subscription options
+        capturedCallback = options.callback;
+        return Promise.resolve({
+          subscriptionId: 'sub-123',
+          unsubscribe: mockUnsubscribe,
+        });
+      });
+      messengerMocks.channelHasSubscription.mockReturnValue(false);
+      messengerMocks.addChannelCallback.mockReturnValue(undefined);
+      messengerMocks.getSelectedAccount.mockReturnValue(mockSelectedAccount);
+
+      await accountActivityService.subscribeAccounts(mockSubscription);
+
+      // Simulate receiving account activity message
+      const activityMessage: AccountActivityMessage = {
+        address: '0x1234567890123456789012345678901234567890',
+        tx: {
+          hash: '0xabc123',
+          chain: 'eip155:1',
+          status: 'confirmed',
+          timestamp: Date.now(),
+          from: '0x1234567890123456789012345678901234567890',
+          to: '0x9876543210987654321098765432109876543210',
+        },
+        updates: [
+          {
+            asset: {
+              fungible: true,
+              type: 'eip155:1/slip44:60',
+              unit: 'ETH',
+            },
+            postBalance: {
+              amount: '1000000000000000000', // 1 ETH
+            },
+            transfers: [
+              {
+                from: '0x1234567890123456789012345678901234567890',
+                to: '0x9876543210987654321098765432109876543210',
+                amount: '500000000000000000', // 0.5 ETH
+              },
+            ],
+          },
+        ],
+      };
+
+      const notificationMessage = {
+        event: 'notification',
+        subscriptionId: 'sub-123',
+        channel:
+          'account-activity.v1.eip155:1:0x1234567890123456789012345678901234567890',
+        data: activityMessage,
+      };
+
+      // Create spy before calling callback to capture publish events
+      const publishSpy = jest.spyOn(messenger, 'publish');
+
+      // Call the captured callback
+      capturedCallback(notificationMessage);
+
+      // Should publish transaction and balance events
+      expect(publishSpy).toHaveBeenCalledWith(
```

Comment:
Hmm nice!
I usually subscribe to this event and see if the test receives it, but I have no preference on mocking the publish either.